src/library/proxytrackmodel.cpp

```diff
@@ -147,26 +155,40 @@ bool ProxyTrackModel::filterAcceptsRow(int sourceRow,
         return false;
     }
 
+    if (m_currentSearch.trimmed().isEmpty()) {
+        return true;
+    }
+
+    QStringList tokens = SearchQueryParser::splitQueryIntoWords(
+            m_currentSearch.trimmed());
+    tokens.removeDuplicates();
+
     const QList<int>& filterColumns = m_pTrackModel->searchColumns();
     QAbstractItemModel* itemModel =
             dynamic_cast<QAbstractItemModel*>(m_pTrackModel);
-    bool rowMatches = false;
-
-    QRegularExpression filter = filterRegularExpression();
-    QListIterator<int> iter(filterColumns);
-
-    while (!rowMatches && iter.hasNext()) {
-        int i = iter.next();
-        QModelIndex index = itemModel->index(sourceRow, i, sourceParent);
-        QVariant data = itemModel->data(index);
-        if (data.canConvert<QString>()) {
-            QString strData = data.toString();
-            if (strData.contains(filter)) {
-                rowMatches = true;
+
+    for (QString token : tokens) {
```

Comment:
```suggestion
    for (const auto& token : std::as_const(tokens)) {
```
Not sure if std::as_const() is strictly required though...